Modules/EMCAL/src/CalibCheck.cxx

```diff
@@ -332,13 +350,242 @@ Quality CalibCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>*
       }
     }
   }
+
+  if (mo->getName() == "NumberNonGoodChannelsFEC" || mo->getName() == "NumberDeadChannelsFEC" || mo->getName() == "NumberBadChannelsFEC") {
+    auto* h = dynamic_cast<TH2*>(mo->getObject());
+    if (h->GetEntries() == 0) {
+      result = Quality::Medium;
+    } else {
+      std::unique_ptr<TH1> h_allsupermod_proj(h->ProjectionX("h_allsupermod_proj"));
+      for (int iFECId = 1; iFECId <= h->GetNbinsX(); iFECId++) {
+        Float_t numChannelsAllSMs = h_allsupermod_proj->GetBinContent(iFECId);
+        if (numChannelsAllSMs > mBadThresholdChannelsFEC) {
+          result = Quality::Bad;
+          break;
+        } else if (numChannelsAllSMs > mMedThresholdChannelsFEC) {
+          result = Quality::Medium;
+        }
+      }
+    }
+  }
+
```

Comment:
We will always have some bad FECs. Consequently the checker will always raise bad. We need to raise an alarm if the number of bad + medium FECs is above a certain level. So you need to determine for each FEC whether it is bad or dead (with the thresholds above), and assign a quality whether the number of FECs is > a certain threshold. @jokonig shall we maybe do this as fraction of FECs per SM?